cli/tests/test_new_command.rs

```diff
@@ -232,12 +222,12 @@ fn test_new_insert_after() {
     ◉  root
     "###);
 
-    // --after cannot be used with --before
-    let stderr = test_env.jj_cmd_cli_error(&repo_path, &["new", "--after", "B", "--before", "D"]);
+    // --after cannot be used with revisions
```

Comment:
In principle, this could put the new commit after B and add D to it as an extra parent. It would still be a breaking change, though, perhaps even more so than it is now.

We can consider doing this later, I suppose (or not -- it's consistent, but it might still be confusing).